src/ng/http.js

```diff
@@ -122,8 +122,12 @@ function $HttpParamSerializerJQLikeProvider() {
                 (topLevel ? '' : ']'));
           });
         } else {
-          parts.push(encodeUriQuery(prefix) + '=' +
-              (toSerialize == null ? '' : encodeUriQuery(serializeValue(toSerialize))));
+          if (isFunction(toSerialize)) {
+            serialize(toSerialize(), prefix);
```

Comment:
Actually, this is not how jQuery does it. They do not recursively serialize a returned object. They just [use the returned value][1].

[1]: https://github.com/jquery/jquery/blob/a6b0705294d336ae2f63f7276de0da1195495363/src/serialize.js#L60-L69